sim/midas/src/main/cc/simif.h

```diff
@@ -21,6 +25,41 @@ double diff_secs(midas_time_t end, midas_time_t start);
 typedef std::map<std::string, size_t> idmap_t;
 typedef std::map<std::string, size_t>::const_iterator idmap_it_t;
 
+/**
+ * Interface for a simulation implementation.
+ *
+ * `simif_t` interacts with the simulation, initialising it, running and and
+ * running finalisation logic at the end.  All simulation implementations
+ * (midasexamples, firesim, bridges, fasedtests) derive this interface,
+ * implementing the driver-specific logic.
+ */
+class simulation_t {
+public:
+  simulation_t(const std::vector<std::string> &args) : args(args) {}
+
+  virtual ~simulation_t() {}
+
+  /**
+   * Simulation main loop.
+   *
+   * @return Exit code to return to the system.
+   */
+  virtual int simulation_run() = 0;
+
+  /**
+   * Simulation initialization. Should set up bridges.
```

Comment:
Be precise about what "setting up bridges" means vis-a-vis mmio -> maybe note it's calling bridge_driver_t::init(), and so there can and will be SW <-> "HW" interactions. 

Speaking of which, we probably need better names than "driver" and "hw" to dilineate between the CPU-hosted and what will be the FPGA-hosted bit of a simulator. Just something to think about.

Reply:
I've adjusted the comments to describe what the tests do now in them. In the future, we should further split these.